src/dense.jl

```diff
@@ -45,6 +45,11 @@ Struct used to select the [Enzyme.jl](https://github.com/EnzymeAD/Enzyme.jl) bac
 
 Defined by [ADTypes.jl](https://github.com/SciML/ADTypes.jl).
 
+!!! warning
+    `AutoEnzyme` can be used by [DifferentiationInterface.jl](https://github.com/JuliaDiff/DifferentiationInterface.jl) to access a restricted subset of Enzyme.jl functionality.
```

Comment:
This feels like the wrong place to have it, since sciml uses ADTypes (created the package), and calls Enzyme directly

Reply:
Yeah but this still feels like the wrong place for this to happen, since ADTypes often directly calls enzyme and doesn’t go through DI.

As mentioned in the other thread I’m working on trying to make the errors throw only if called within DI

Reply:
I thought this would be a good addition for the many packages who do call Enzyme through DI, without hurting the others, but I'll leave it up to you. Feel free to revive this PR if you change your mind, I'm setting it as a draft for the time being.